src/hooks/useApi.ts

```diff
@@ -1,9 +1,136 @@
 import { useAppContext } from "../store/app.context";
+import * as API from "../api";
+import data from "../data.json";
+import { Employee } from "../models";
 
 export default function useApi() {
-    const appContext = useAppContext();
+  const appContext = useAppContext();
 
-    function getEmployees() {
-        
+  async function getEmployees() {
+    try {
+      const employees = await API.getEmployees();
+      appContext.dispatch({ type: "SET_EMPLOYEES", payload: employees });
+      return employees;
+    } catch (error: any) {
+      appContext.showToast({
+        message: "There was en error while fetching employees",
```

Comment:
Move all error text messages in the app to a separate file.

Reply:
done